internal/hammer/diff.go

```diff
@@ -428,8 +436,6 @@ func (g *Generator) columnTypeEqual(x, y spansql.ColumnDef) bool {
 func (g *Generator) constraintEqual(x, y spansql.TableConstraint) bool {
 	x.Position = spansql.Position{}
 	y.Position = spansql.Position{}
-	x.ForeignKey.Position = spansql.Position{}
-	y.ForeignKey.Position = spansql.Position{}
```

Comment:
If you delete the position update here, it will be treated as a difference when you change only the position of the definition.

```
CREATE TABLE t2 (
  t2_1 INT64 NOT NULL,
  t2_2 INT64 NOT NULL,
  CONSTRAINT FK_t2 FOREIGN KEY (t2_1) REFERENCES t1 (t1_1),
) PRIMARY KEY(t2_1);
```

```
CREATE TABLE t2 (
  t2_1 INT64 NOT NULL,
  t2_2 INT64 NOT NULL,

  CONSTRAINT FK_t2 FOREIGN KEY (t2_1) REFERENCES t1 (t1_1),
) PRIMARY KEY(t2_1);
```

Reply:
Fixed and add a test case about it.
BTW, this PR supports only the foreign key's constraint, not the check constraint.
PTAL 🙏

Reply:
Thank you for correcting and adding the test!
 I understood about the check constraint.